### PR TITLE
Get User Posts

### LINKED ISSUE #154: Get user homepage
### Story
- Currently, the user homepage is not implemented. We need to implement an endpoint to gel all posts a user created to be added to its homepage.

### Tasks
- [ ] To be implemented. 

### Acceptance criteria
- [ ] Whenever a user hits the needed endpoint, it responds with all posts he/she created.

### PR DESCRIPTION
**Pull request by MHamada96**
Closes #154 
- added a /users/{{user_id}}/posts route that will return user posts
- make getUserPosts Function inside posts service
- create a validator for the user uuid parameter
- create a getUserPosts function inside the posts Repo ( a copy of getAllPosts the extra condition)

### CODE REVIEW

File: examples/examples.http [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
+
+
+@token = eyJhbGciOiJSUzI1NiIsImtpZCI6IjM2NGU4NTQ1NzI5OWQ5NzIxYjczNDQyZGNiNTQ3Y2U2ZDk4NGRmNTkiLCJ0eXAiOiJKV1QifQ.eyJuYW1lIjoib21hciBzYW15IiwicGljdHVyZSI6Imh0dHBzOi8vbGgzLmdvb2dsZXVzZXJjb250ZW50LmNvbS9hLS9BT2gxNEdoYlBIaGpFNGpveVZzaGhOSEt2aFdCQ0V4cVYyZWtfenpPNThlTlBRPXM5Ni1jIiwiaXNzIjoiaHR0cHM6Ly9zZWN1cmV0b2tlbi5nb29nbGUuY29tL3BpY2tpZnktMTM4ZjkiLCJhdWQiOiJwaWNraWZ5LTEzOGY5IiwiYXV0aF90aW1lIjoxNjI5NTQwMDI2LCJ1c2VyX2lkIjoiVTBhMTdWdzFVUk5IQXVPMHE3S1hnNkZwbmdpMSIsInN1YiI6IlUwYTE3VncxVVJOSEF1TzBxN0tYZzZGcG5naTEiLCJpYXQiOjE2Mjk1NDUyNjAsImV4cCI6MTYyOTU0ODg2MCwiZW1haWwiOiJvbWFyc2FteTA5QGdtYWlsLmNvbSIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJmaXJlYmFzZSI6eyJpZGVudGl0aWVzIjp7Imdvb2dsZS5jb20iOlsiMTAwMzIyNzg1NzkyNzk3NTY4ODY1Il0sImVtYWlsIjpbIm9tYXJzYW15MDlAZ21haWwuY29tIl19LCJzaWduX2luX3Byb3ZpZGVyIjoiZ29vZ2xlLmNvbSJ9fQ.JmM1YCr6Ywk0GxKC9qTcixJZHnVpTwjX87hQOEuInorf5fwrQEqlK4zLELk1Jj4gQ0xq6HFfb7LJHFIwRwvwVNsWLCWecLwJPEv0UZzy6O2fP23rRUQwIfkW2vqZkoR3b2rHy3j3-pfCRN09wjhPh0YAm9QgSlEJn8upg0AmvJvBTWP_4-6SqOWx_QN81OMzMB5oq33dwJOKcX0KpsfUyJJfRdGr7d5QXG5C1SEb6oFUJjAQETZnDXmhj6uaMOL1PTGJ3PCATqaxevIi6dbQ403epjZrw7FdAKFAI26HfZwgXYVd3CDDFy2qvfkWdL8sQW9v6-4sA-vpmEEI8q8bFg
```
Comment (Nilegfx) [inline examples/examples.http]: image

Comment (Nilegfx) [reply examples/examples.http]: @AhmedAdel999 shayef zmelak by3mel eh? kolkom sabakeen!

File: openAPI/post.openAPI.json [hunk 2]
```diff
@@ -83,6 +83,39 @@
         }
       }
     },
+    "/posts/homepage/{userid}": {
```
Comment (Nilegfx) [inline openAPI/post.openAPI.json]: `/users/{userid}`
`/users/{userid}/posts`

`/users/{userid}`
```js
{
...profileData
}
```

`/users/{userid}/posts`
```js
{
userPosts
}
```

File: openAPI/post.openAPI.json [hunk 3]
```diff
@@ -83,6 +83,39 @@
...
+      "get": {
+        "tags": ["posts"],
```
Comment (Nilegfx) [inline openAPI/post.openAPI.json]: I guess this should be ["users"]

File: examples/examples.http (hunk 1 above)
Comment (MHamada96) [reply examples/examples.http]: Sorry! updated 😂

File: openAPI/post.openAPI.json (hunk 3 above)
Comment (MHamada96) [reply openAPI/post.openAPI.json]: Yep! we have updated it and added a new user's route. the swagger file was updated as well.

**Review by Nilegfx — DISMISSED** [review]
didn't review it deeply. so it is ok to expect some inconveniences after merge. this took too much.

**Review by Nilegfx — APPROVED** [review]
LGTM

**Pull request merged**